nuvla/job_engine/job/version.py

```diff
@@ -1,14 +1,70 @@
 import importlib_metadata
 import logging
 import os
+import re
 
 logger = logging.getLogger(__name__)
 
-package_name = "nuvla-job-engine"
-version = os.getenv('JOB_ENGINE_VERSION')
-if not version:
+def _get_version_from_env():
+    return os.getenv('JOB_ENGINE_VERSION')
+
+def _get_version_from_package():
     try:
-        version = importlib_metadata.version(package_name)
+        return importlib_metadata.version("nuvla-job-engine")
     except importlib_metadata.PackageNotFoundError:
         logger.warning('Cannot retrieve Job-engine version')
-        version = None
+        return None
+
+def _version_to_tuple(engine_version_str: str) -> tuple:
+    ver_trim_re = re.compile(r'\.?[^.0-9]+[0-9]*$')
```

Comment:
I think this line will benefit from staying outside of the function to prevent the regex to be recompiled every time.